R/LSBuffonsneedlemanipulation.R

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (C) 2019 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+LSBuffonsneedlemanipulation   <- function(jaspResults, dataset, options, state = NULL){
+  .buffonsNeedleManipulationSummaryTable(jaspResults, options)
+  .buffonsNeedleManipulationPropDistPlot(jaspResults, options)
+  .buffonsNeedleManipulationPiDistPlot(jaspResults, options)
+  return()
+}
+
+
+
+.buffonsNeedleManipulationSummaryTable <- function(jaspResults, options){
+  if(!is.null(jaspResults[["summaryTable"]])) return()
+  # example d for computation
+  d <- 5
+  l <- options[["length"]]*d/100
+  
+  ## Summary Table
+  summaryTable <- createJaspTable(title = gettext("Summary Table"))
+  summaryTable$position <- 1
+  summaryTable$dependOn(c("k", "n", "a", "b", "length", "CI"))
+  #summaryTable$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
+  
+  summaryTable$addColumnInfo(name = "NumCrosses", title = gettext("Crosses"), type = "integer")
+  summaryTable$addColumnInfo(name = "NumObservations", title = gettext("Observations"), type = "integer")
+  summaryTable$addColumnInfo(name = "Median", title = gettextf("Median for %s", "\u03c0"),   type = "number")
+  summaryTable$addColumnInfo(name = "lowerCI", title = gettext("Lower"), type = "number", 
+                            overtitle = gettext(paste0(options[["CI"]]*100, "% Credible Interval")))
+  summaryTable$addColumnInfo(name = "upperCI", title = gettext("Upper"), type = "number", 
+                            overtitle = gettext(paste0(options[["CI"]]*100, "% Credible Interval"))) 
```

Comment:
same - gettextf

Reply:
what about `gettext(paste0(options[["CI"]]*100, "% CI: ", "[",CI95lower,", ", CI95upper, "]"))`?

I cannot find any instruction in google about gettextf(). Could you send me sth about that?

Reply:
see https://stat.ethz.ch/R-manual/R-devel/library/base/html/sprintf.html and https://www.gastonsanchez.com/r4strings/c-style-formatting.html

`gettextf` is defined as:
```r
function (fmt, ..., domain = NULL) {
  sprintf(gettext(fmt, domain = domain), ...)
}
```
so you can insert characters with `%s` and e.g., integers with `%d`. It's recommended to insert everything as character with `%s` though. If there are multiple things inserted, it's recommended to specify the order. For example:
```r
> gettextf("%2$s %1$s", "a", "b")
[1] "b a"
```
that way, translators can reorder sentences as well.


> what about `gettext(paste0(options[["CI"]]*100, "% CI: ", "[",CI95lower,", ", CI95upper, "]"))`

No. This won't work because then the translators might get this to translate:
```
paste0(options[["CI"]]*100, "% CI: ", "[",CI95lower,", ", CI95upper, "]")
```
so they'd be translating literal R code... which is not ideal because if they mess up then you might have a bug, but in one language only. In general, there should only be string literals in `gettext` and if you need to construct a string programmatically you always need `gettextf`.

Reply:
so I wrote sth like `gettextf("%1$s%2$s%3$s%4$s%5$s%6$s", options[["CI"]]*100, "% CI: [", CI95lower, ",", CI95upper, "]")
`
the output looks good.

Reply:
`[1] "95% CI: [1,2]"`

Reply:
Right but string literals don't need to be put in like that...
This should be something like:
```r
gettextf("%s%% CI [%s, %s]", options[["CI"]]*100, CI95lower, CI95upper)
```
with `gettextf("%1$s%2$s%3$s%4$s%5$s%6$s", options[["CI"]]*100, "% CI: [", CI95lower, ",", CI95upper, "]")`, you're asking translators to translate `"%1$s%2$s%3$s%4$s%5$s%6$s"`...